test/new-e2e/tests/agent-metric-logs/log-agent/k8s-logs/file_tailing_test.go

```diff
@@ -0,0 +1,174 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package k8sfiletailing
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"slices"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments"
+	kindfilelogger "github.com/DataDog/datadog-agent/test/new-e2e/tests/agent-metric-logs/kindfilelogging"
+)
+
+type k8sSuite struct {
+	e2e.BaseSuite[environments.Kubernetes]
+}
+
+func TestK8sSuite(t *testing.T) {
+	e2e.Run(t, &k8sSuite{}, e2e.WithProvisioner(kindfilelogger.Provisioner()))
+}
+
+func (v *k8sSuite) TestSingleLogAndMetadata() {
+	v.Env().FakeIntake.Client().FlushServerAndResetAggregators()
+	var backOffLimit int32 = 4
+	testLogMessage := "Test log message"
+
+	jobSpcec := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "job-1",
+			Namespace: "default",
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:    "query-job-1",
+							Image:   "ubuntu",
+							Command: []string{"echo", testLogMessage},
+						},
+					},
+					RestartPolicy: corev1.RestartPolicyNever,
+				},
+			},
+			BackoffLimit: &backOffLimit,
+		},
+	}
+
+	_, err := v.Env().KubernetesCluster.Client().BatchV1().Jobs("default").Create(context.TODO(), jobSpcec, metav1.CreateOptions{})
+	assert.NoError(v.T(), err, "Could not properly start job")
+
+	v.EventuallyWithT(func(c *assert.CollectT) {
+		logsServiceNames, err := v.Env().FakeIntake.Client().GetLogServiceNames()
+		assert.NoError(c, err, "Error starting job")
+		assert.Contains(c, logsServiceNames, "ubuntu", "Ubuntu service not found")
```

Comment:
similar to this comment:
https://github.com/DataDog/datadog-agent/pull/24529/commits/d636759c2240acfdcc78ddf332654c34269964dd#r1583443829

Reply:
done!